src/util.ts

```diff
@@ -77,3 +78,26 @@ export function isJobStatusSuccess() {
 
   return jobStatus === 'success';
 }
+
+export function isGhes(): boolean {
+  const ghUrl = new URL(process.env['GITHUB_SERVER_URL'] || 'https://github.com');
+  return ghUrl.hostname.toUpperCase() !== 'GITHUB.COM';
+}
+
+export function isCacheFeatureAvailable(): boolean {
+  if (!cache.isFeatureAvailable()) {
+    if (isGhes()) {
+      throw new Error(
+        'Caching is only supported on GHES version >= 3.5. If you are on a version >= 3.5, please check with your GHES admin if the Actions cache service is enabled or not.'
+      );
+    } else {
+      core.warning(
+        'This runner is not configured to use the cache service. Caching will be skipped'
```

Comment:
Can we update to the same thing from https://github.com/actions/setup-python/pull/363?

```suggestion
        'The runner was not able to contact the cache service. Caching will be skipped'
```